### ISSUE TITLE
Spots aren't being retrieved

### ISSUE THREAD

**Issue by notian85** [spotweb/spotweb#823]
**Describe the bug/issue**
Spots are never retrieved, screen stays blank.

**Have you searched the internet or Github for an answer?**
Yes, endlessly. Tried with install of postgresql and without. Reinstalled a few times and changed setting during install.

**To Reproduce**
See following log:
2023-07-04 2:18:47
Readiness probe failed: HTTP probe failed with statuscode: 500
2023-07-04 2:18:46
Created container postgres
2023-07-04 2:18:46
Started container postgres
2023-07-04 2:18:46
Readiness probe failed: Get "http://172.16.0.166:8000/readyz": dial tcp 172.16.0.166:8000: connect: connection refused
2023-07-04 2:18:44
Started container bootstrap-controller
2023-07-04 2:18:44
Container image "ghcr.io/cloudnative-pg/postgresql:15.2" already present on machine
2023-07-04 2:18:43
Created container bootstrap-controller
2023-07-04 2:18:42
Add eth0 [172.16.0.166/16] from ix-net
2023-07-04 2:18:42
Container image "tccr.io/truecharts/cloudnative-pg:v1.20.0@sha256:b226152774c438d4826d29b63efb559a4d99f7a4890cae26a5e49e88f669c38c" already present on machine
2023-07-04 2:18:41
Job completed
2023-07-04 2:18:41
Successfully assigned ix-spotweb/spotweb-cnpg-main-2 to ix-truenas
2023-07-04 2:18:41
Startup probe failed: dial tcp 172.16.0.160:80: connect: connection refused
2023-07-04 2:18:30
Started container spotweb
2023-07-04 2:18:30
Created container join
2023-07-04 2:18:30
Started container join
2023-07-04 2:18:29
Container image "ghcr.io/cloudnative-pg/postgresql:15.2" already present on machine
2023-07-04 2:18:29
Created container spotweb
2023-07-04 2:18:28
Container image "tccr.io/truecharts/spotweb:v1.5.4@sha256:be432498a6b9fd6b8a447bfe5e841791a4fcc4a8de829bbc054df73714c91296" already present on machine
2023-07-04 2:18:28
Created container bootstrap-controller
2023-07-04 2:18:28
Started container bootstrap-controller
2023-07-04 2:18:26
Add eth0 [172.16.0.165/16] from ix-net
2023-07-04 2:18:26
Container image "tccr.io/truecharts/cloudnative-pg:v1.20.0@sha256:b226152774c438d4826d29b63efb559a4d99f7a4890cae26a5e49e88f669c38c" already present on machine
2023-07-04 2:18:25
Successfully assigned ix-spotweb/spotweb-cnpg-main-2-join-v25h2 to ix-truenas
2023-07-04 2:18:23
0/1 nodes are available: 1 pod has unbound immediate PersistentVolumeClaims. preemption: 0/1 nodes are available: 1 Preemption is not helpful for scheduling.
2023-07-04 2:18:23
Successfully provisioned volume pvc-d5b2f863-8e97-4cd9-9e5d-8a6df224c4db
2023-07-04 2:18:23
Successfully provisioned volume pvc-950841e1-b8c9-475e-b130-1ea69ea9c77e
2023-07-04 2:18:21
Creating instance spotweb-cnpg-main-2
2023-07-04 2:18:21
0/1 nodes are available: 1 persistentvolumeclaim "spotweb-cnpg-main-2-wal" not found. preemption: 0/1 nodes are available: 1 Preemption is not helpful for scheduling.
2023-07-04 2:18:21
External provisioner is provisioning volume for claim "ix-spotweb/spotweb-cnpg-main-2"
2023-07-04 2:18:21
waiting for a volume to be created, either by external provisioner "zfs.csi.openebs.io" or manually created by system administrator
2023-07-04 2:18:21
Created pod: spotweb-cnpg-main-2-join-v25h2
2023-07-04 2:18:21
External provisioner is provisioning volume for claim "ix-spotweb/spotweb-cnpg-main-2-wal"
2023-07-04 2:18:21
waiting for a volume to be created, either by external provisioner "zfs.csi.openebs.io" or manually created by system administrator
2023-07-04 2:18:13
Successfully pulled image "ghcr.io/cloudnative-pg/pgbouncer:1.18.0" in 30.140716649s
2023-07-04 2:18:13
Created container pgbouncer
2023-07-04 2:18:13
Created container pgbouncer
2023-07-04 2:18:13
Started container pgbouncer
2023-07-04 2:18:13
Started container pgbouncer
2023-07-04 2:18:13
Readiness probe failed: dial tcp 172.16.0.161:5432: connect: connection refused
2023-07-04 2:18:13
Readiness probe failed: dial tcp 172.16.0.162:5432: connect: connection refused
2023-07-04 2:18:12
Successfully pulled image "ghcr.io/cloudnative-pg/pgbouncer:1.18.0" in 30.015766054s
2023-07-04 2:18:07
Readiness probe failed: HTTP probe failed with statuscode: 500
2023-07-04 2:18:06
Created container postgres
2023-07-04 2:18:06
Started container postgres
2023-07-04 2:18:06
Readiness probe failed: Get "http://172.16.0.164:8000/readyz": dial tcp 172.16.0.164:8000: connect: connection refused
2023-07-04 2:18:04
Container image "ghcr.io/cloudnative-pg/postgresql:15.2" already present on machine
2023-07-04 2:18:03
Created container bootstrap-controller
2023-07-04 2:18:03
Started container bootstrap-controller
2023-07-04 2:18:02
Add eth0 [172.16.0.164/16] from ix-net
2023-07-04 2:18:02
Container image "tccr.io/truecharts/cloudnative-pg:v1.20.0@sha256:b226152774c438d4826d29b63efb559a4d99f7a4890cae26a5e49e88f669c38c" already present on machine
2023-07-04 2:18:00
Job completed
2023-07-04 2:18:00
Successfully assigned ix-spotweb/spotweb-cnpg-main-1 to ix-truenas
2023-07-04 2:17:50
Created container initdb
2023-07-04 2:17:50
Started container initdb
2023-07-04 2:17:48
Container image "ghcr.io/cloudnative-pg/postgresql:15.2" already present on machine
2023-07-04 2:17:47
Created container bootstrap-controller
2023-07-04 2:17:47
Started container bootstrap-controller
2023-07-04 2:17:45
Add eth0 [172.16.0.163/16] from ix-net
2023-07-04 2:17:45
Container image "tccr.io/truecharts/cloudnative-pg:v1.20.0@sha256:b226152774c438d4826d29b63efb559a4d99f7a4890cae26a5e49e88f669c38c" already present on machine
2023-07-04 2:17:44
Successfully assigned ix-spotweb/spotweb-cnpg-main-1-initdb-g6kfr to ix-truenas
2023-07-04 2:17:42
Created container spotweb-system-cnpg-wait
2023-07-04 2:17:42
Started container bootstrap-controller
2023-07-04 2:17:42
Started container bootstrap-controller
2023-07-04 2:17:42
Started container spotweb-system-cnpg-wait
2023-07-04 2:17:42
Pulling image "ghcr.io/cloudnative-pg/pgbouncer:1.18.0"
2023-07-04 2:17:42
Pulling image "ghcr.io/cloudnative-pg/pgbouncer:1.18.0"
2023-07-04 2:17:41
Add eth0 [172.16.0.161/16] from ix-net
2023-07-04 2:17:41
Container image "tccr.io/truecharts/cloudnative-pg:v1.20.0@sha256:b226152774c438d4826d29b63efb559a4d99f7a4890cae26a5e49e88f669c38c" already present on machine
2023-07-04 2:17:41
Add eth0 [172.16.0.162/16] from ix-net
2023-07-04 2:17:41
Container image "tccr.io/truecharts/cloudnative-pg:v1.20.0@sha256:b226152774c438d4826d29b63efb559a4d99f7a4890cae26a5e49e88f669c38c" already present on machine
2023-07-04 2:17:41
Successfully provisioned volume pvc-d5c7ed8b-8993-4a01-a73f-87ca23bdf437
2023-07-04 2:17:41
0/1 nodes are available: 1 pod has unbound immediate PersistentVolumeClaims. preemption: 0/1 nodes are available: 1 Preemption is not helpful for scheduling.
2023-07-04 2:17:41
Successfully provisioned volume pvc-6ae631b8-d790-4645-af3e-c42028331fe6
2023-07-04 2:17:41
Updated LoadBalancer with new IPs: [] -> [192.168.1.166]
2023-07-04 2:17:41
Created container bootstrap-controller
2023-07-04 2:17:41
Created container bootstrap-controller
2023-07-04 2:17:40
Add eth0 [172.16.0.160/16] from ix-net
2023-07-04 2:17:40
Container image "tccr.io/truecharts/db-wait-postgres:1.1.0@sha256:a163c7836d7bb436a428f5d55bbba0eb73bcdb9bc202047e2523bbb539c113e6" already present on machine
2023-07-04 2:17:36
Created pod: spotweb-cnpg-main-1-initdb-g6kfr
2023-07-04 2:17:36
Successfully assigned ix-spotweb/spotweb-cnpg-main-rw-7c4bf5dc6d-p5588 to ix-truenas
2023-07-04 2:17:35
Job completed
2023-07-04 2:17:35
Ensuring load balancer
2023-07-04 2:17:35
Scaled up replica set spotweb-6d74f8487c to 1
2023-07-04 2:17:35
Created pod: spotweb-6d74f8487c-r2lk2
2023-07-04 2:17:35
Applied LoadBalancer DaemonSet kube-system/svclb-spotweb-33331014
2023-07-04 2:17:35
Successfully assigned ix-spotweb/spotweb-6d74f8487c-r2lk2 to ix-truenas
2023-07-04 2:17:35
Creating PodDisruptionBudget spotweb-cnpg-main-primary
2023-07-04 2:17:35
Creating ServiceAccount
2023-07-04 2:17:35
No matching pods found
2023-07-04 2:17:35
Creating Cluster Role
2023-07-04 2:17:35
Primary instance (initdb)
2023-07-04 2:17:35
External provisioner is provisioning volume for claim "ix-spotweb/spotweb-cnpg-main-1"
2023-07-04 2:17:35
External provisioner is provisioning volume for claim "ix-spotweb/spotweb-cnpg-main-1-wal"
2023-07-04 2:17:35
0/1 nodes are available: 1 pod has unbound immediate PersistentVolumeClaims. preemption: 0/1 nodes are available: 1 Preemption is not helpful for scheduling.
2023-07-04 2:17:35
Scaled up replica set spotweb-cnpg-main-rw-7c4bf5dc6d to 2
2023-07-04 2:17:35
Created pod: spotweb-cnpg-main-rw-7c4bf5dc6d-wtkjw
2023-07-04 2:17:35
Created pod: spotweb-cnpg-main-rw-7c4bf5dc6d-p5588
2023-07-04 2:17:35
Successfully assigned ix-spotweb/spotweb-cnpg-main-rw-7c4bf5dc6d-wtkjw to ix-truenas
2023-07-04 2:17:35
waiting for a volume to be created, either by external provisioner "zfs.csi.openebs.io" or manually created by system administrator
2023-07-04 2:17:35
waiting for a volume to be created, either by external provisioner "zfs.csi.openebs.io" or manually created by system administrator
2023-07-04 2:17:29
Started container spotweb-manifests
2023-07-04 2:17:28
Add eth0 [172.16.0.158/16] from ix-net
2023-07-04 2:17:28
Container image "tccr.io/truecharts/kubectl:v1.26.0@sha256:323ab7aa3e7ce84c024df79d0f364282c1135499298f54be2ade46508a116c4b" already present on machine
2023-07-04 2:17:28
Created container spotweb-manifests
2023-07-04 2:17:27
Created pod: spotweb-manifests-pvmjv
2023-07-04 2:17:27
Successfully assigned ix-spotweb/spotweb-manifests-pvmjv to ix-truenas

**Expected behavior**
retrieving the spots

**Screenshots**
none

**Desktop (please complete the following information):**
 - OS: TrueNAS-SCALE-22.12.3.1
 - Browser any browser
 - Spotweb Version spotweb
1.5.4_8.0.22
 - Which branch  stable
 -  PHP version don't know
 -  Device information Lenovo M710s I5-7500

**Comment by notian85** [comment]
I installed it via the truecharts catalogue: https://truecharts.org/charts/stable/spotweb/

**Comment by Sweepr** [comment]
Have you configured it properly?
Did you run "retrieve.php?

**Comment by Sweepr** [comment]
Never heard of this package before, is it possible to ask the creator for help because we do not know the exact condition underwhich it is installed and it seems to be somekind of container/dockerish thing?

**Comment by Sweepr** [comment]
This is a clue "Readiness probe failed: Get "http://172.16.0.166:8000/readyz": dial tcp 172.16.0.166:8000: connect: connection refused"

Not sure what it means tho.

**Comment by notian85** [comment]
Hi Sweepr,

A few hours later to when I wrote this the spots appeared, but only from installation moment onwards. So I guess it just needed some time to retrieve, however then the new problem that the old spots wouldn't retrieve was there.

I figured that out as well, it seems to be a more common issue. Dr. Frankenstein has written a guide about ithttps://drfrankenstein.co.uk/spotweb-not-backfilling/

This was exactly the issue I had. I used on Truenas an app called adminer and found the database location, name and password using the following guide: https://truecharts.org/manual/SCALE/guides/migration-pvc/#mariadb-databases. Then I did step 3 onwards exactly as in the Frankenstein guide and the spots appeared. I never figured out how to run retrieve.php, it is in a docker container and not sure how to run it. You probably have to mount the PVC etc.
